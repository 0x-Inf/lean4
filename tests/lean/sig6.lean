variables A B : (Type U)
variables t1 t2 : A ⨯ B
axiom pair_Ax {A : (Type U)} {B : A → (Type U)} (p : sig x, B x) : (tuple (sig x, B x) : (proj1 p), (proj2 p)) = p
theorem spairext {A B : (Type U)} {p1 p2 : A ⨯ B} (H1 : proj1 p1 = proj1 p2) (H2 : proj2 p1 = proj2 p2) : p1 = p2
:= calc p1  = tuple (proj1 p1), (proj2 p1)  :  symm (pair_Ax p1)
        ... = tuple (proj1 p2), (proj2 p1)  :  { H1 }
        ... = tuple (proj1 p2), (proj2 p2)  :  { H2 }
        ... = p2                            :  pair_Ax p2