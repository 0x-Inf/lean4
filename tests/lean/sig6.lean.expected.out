  Set: pp::colors
  Set: pp::unicode
  Assumed: A
  Assumed: B
  Assumed: t1
  Assumed: t2
  Assumed: pair_Ax
  Proved: spairext
